src/app/code/community/Shopgate/Cloudapi/Model/OAuth2/Db/Pdo.php

```diff
@@ -222,6 +222,29 @@ public function setAuthorizationCode(
         return $stmt->execute(compact('code', 'clientId', 'userId', 'resourceType', 'expires', 'scope'));
     }
 
+    /**
+     * @param string $oldEmail
+     * @param string $newEmail
+     */
+    public function updateEmailTokens($oldEmail, $newEmail)
+    {
+        $stmt = $this->db->prepare(
+            $sql = sprintf(
```

Comment:
Unfortunately, updating the token entry will automatically invalidate the access token as the expires column has a "on update CURRENT_TIMESTAMP".

Reply:
hm, a few solutions that I can think of at the moment:
1) see if we can switch the using customer id's, don't think it was possible when I was implementing. I think the oauth library interface did not allow much play with that.
2) clone, delete, add token - something feels weird about this one though, like we are doing something dirty.